types/defines/vectorize.d.ts

```diff
@@ -127,15 +142,19 @@ interface VectorizeVectorMutation {
 }
 
 /**
-* Results of an operation that performed a mutation on a set of vectors
-* with the v2 version of Vectorize.
-* Here, `mutationId` is the identifier for the last mutation processed by Vectorize.
-*/
-interface VectorizeVectorMutationV2 {
-  /* The identifier for the last mutation processed by Vectorize. */
+ * Result type indicating a mutation on the Vectorize Index.
+ * Actual mutations are processed async where the `mutationId` is the unique identifier for the operation.
+ */
+interface VectorizeAsyncMutation {
+  /** The unique identifier for the async mutation operation containing the changeset. */
   mutationId: string;
 }
 
+/**
+ * A Vectorize Vector Search Index for querying vectors/embeddings.
+ *
+ * @deprecated This is a legacy version from beta version Vectorize.
```

Comment:
This should not be deprecated yet, since V2 is not yet released and stable

Reply:
Replaced with a warning that it will be soon
https://github.com/cloudflare/workerd/pull/2421/files#diff-4386ee8e0b9580e2b88770e50a5e19d669bd8d8957df5e215b633330a9f2cd23R135